script.js

```diff
@@ -1,4 +1,421 @@
+    gameMode = input;
+    outputValue = `You have selected the ${gameMode} game mode. <br>`;
+    if (input == ARCADE) {
+      generateScoreboard();
+      gameState = ROLL;
+      outputValue += `Player ${currentPlayer} will roll first.`;
+    } else {
+      generateFixtures();
+      gameState = ROLL_PLAYER_ONE;
+      outputValue += `The first fixture is Player ${playerOne} VS Player ${playerTwo}.<br>Player ${playerOne} will roll first.`;
+    }
+    return outputValue;
+  } else {
+    outputValue = `Invalid entry. Please input either ${KNOCKOUT} or ${ARCADE}.`;
+    return outputValue;
+  }
+};
+
+// Helper Function to Select Low/High
+var setLowHigh = function (input) {
+  if (input == LOW || input == HIGH) {
+    lowHigh = input;
+    outputValue = `You have selected that the ${lowHigh}EST roll wins.`;
+    gameState = AUTO_CHOOSE;
+    return outputValue;
+  } else {
+    outputValue = `Invalid entry. Please input either ${LOW} or ${HIGH}.`;
+    return outputValue;
+  }
+};
+
+// Helper Function to Combine Integers in Array
+var combineArray = function (array) {
+  var arrayLength = array.length;
+  var outputValue = 0;
+  for (var counter = 0; counter < arrayLength; counter += 1) {
+    outputValue = 10 * outputValue + array[counter];
+  }
+  return outputValue;
+};
+
+// Helper Function to Automatically Arrange Array.
+var autoChooseFunction = function (array, lowHigh) {
+  if (lowHigh == LOW) {
+    var outputArray = array.sort(function (a, b) {
+      return a - b;
+    });
+    var outputNumber = combineArray(outputArray);
+    return outputNumber;
+  } else {
+    var outputArray = array.sort(function (a, b) {
+      return b - a;
+    });
+    var outputNumber = combineArray(outputArray);
+    return outputNumber;
+  }
+};
+
+//Helper Function to Activate Auto Choose Function
+var setAutoChoose = function (input) {
+  if (input == "Yes") {
+    autoChoose = true;
+    var outputValue = `You have selected to use the auto choose function`;
+    gameState = MODE;
+  } else if (input == "No") {
+    autoChoose = false;
+    var outputValue = `You have selected to not use the auto choose function`;
+    gameState = MODE;
+  } else {
+    var outputValue = `Invalid input. Please input either "Yes" or "No" to the use of an automatic dice order selector.`;
+  }
+  return outputValue;
+};
+
+// Helper Function to Generate Overall Scoreboard with Each Player Representing an Entry in List
+var generateScoreboard = function () {
+  globalScoreboard = Array(numPlayers).fill(0);
+  console.log(globalScoreboard);
+};
+
+// Helper Function to Generate Fixtures into Global Variables currentFixtures and nextFixtures
+var generateFixtures = function () {
+  var playerList = Array(numPlayers)
+    .fill()
+    .map((_, i) => i + 1);
+  var randomizedFixture = playerList.sort(() => Math.random() - 0.5);
+  // Check how many players get a pass to the next round
+  var numberOfPasses = 2 ** Math.ceil(Math.log2(numPlayers)) - numPlayers;
```

Comment:
I like how you managed to implement the array methods you found online in your own code